source/contributed/caching-overview.md

```diff
@@ -0,0 +1,76 @@
+
+## Datastores
+
+### Memory
+
+The most common place to store cached data is in [Memory](/global-objects.html#Memory-object). This is the only option for true persistance- anything stored in [Memory](/global-objects.html#Memory-object) is going to stay there until removed, so if a value is extremely expensive or most be saved this is the place to put it.
+
+There are two major drawbacks [Memory](/global-objects.html#Memory-object)-
+
+*   Space in [Memory](/global-objects.html#Memory-object) is limited to 2048kb of space.
+*   JSON.parse is run on the [Memory](/global-objects.html#Memory-object) string each tick it is accessed, which is more expensive the more data that is stored.
+
+For these reasons it makes sense to limit what is placed in [Memory](/global-objects.html#Memory-object).
+
+
+### Global
+
+The [game loop]('http://support.screeps.com/hc/en-us/articles/204825672-New-main-loop-architecture') archetecture allows you to define a "loop" function which gets run each tick while allowing you to define expensive one-time-run code in the outer scope. This is most commonly used to `require` modules-
+
+    // executed on new global
+    var mod = require('mod');
+
+    module.exports.loop = function() {
+        // executed every tick
+        mod.foo();
+    }
+
+Another example shows how this can be utilized for storing the results of an expensive function. After the first time it the function is called the variable will be populated for future calls:
+
+    let runExpensiveCodeResults = false
+    function runExpensiveCode() {
+        if(!runExpensiveCodeResults) {
+            runExpensiveCodeResults = someExpensiveCode()
+        }
+        return runExpensiveCodeResults
+    }
+
+These two examples have a drawback in that they are only run or defined when the code is first loaded. This can be bypassed by accessing the `[global](https://nodejs.org/api/globals.html#globals_global)` object, which is a special object in Node that is accessible everywhere.
+
+    function runExpensiveCode() {
+        if(!global.runExpensiveCodeResults) {
+            global.runExpensiveCodeResults = someExpensiveCode()
+        }
+        return global.runExpensiveCodeResults
+    }
+
+
+There are some severe limits to this-
+
+*   Each worker has it's own `global` state, and code will bounce between at least four nodes at a time. This means data placed in global one tick will only have a 25% chance of appear next (although it can be added again until it's on all four nodes, bringing the change up to 100%) and that deleted data could easily show up again in the next tick.
```

Comment:
In my opinion this should be a little less specific, while four nodes might be normal right now this might change in the future (or on private servers). I would rephrase it to something along the lines of "code execution might get moved between multiple workers in between ticks, therefor you shouldn't expect your data to always be available as it might not be cached on the new worked yet".

Anyways your second sentence sounds confusing right now and might need some rephrasing if you disagree with my suggestion.